services/operations.go

```diff
@@ -0,0 +1,430 @@
+					Currency: OasisCurrency,
+				},
+				RelatedOperations: []*types.OperationIdentifier{
+					&types.OperationIdentifier{
+						Index: opIndex,
+					},
+				},
+			},
+		)
+	case staking.MethodBurn:
+		var body staking.Burn
+		if err := cbor.Unmarshal(m.tx.Body, &body); err != nil {
+			return fmt.Errorf("malformed body: %w", err)
+		}
+
+		m.ops = append(m.ops,
+			&types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: opIndex,
+				},
+				Type:   OpBurn,
+				Status: m.status,
+				Account: &types.AccountIdentifier{
+					Address: m.txSignerAddress,
+				},
+				Amount: &types.Amount{
+					Value:    "-" + body.Amount.String(),
+					Currency: OasisCurrency,
+				},
+			},
+		)
+	case staking.MethodAddEscrow:
+		var body staking.Escrow
+		if err := cbor.Unmarshal(m.tx.Body, &body); err != nil {
+			return fmt.Errorf("malformed body: %w", err)
+		}
+
+		m.ops = append(m.ops,
+			&types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: opIndex,
+				},
+				Type:   OpTransfer,
+				Status: m.status,
+				Account: &types.AccountIdentifier{
+					Address: m.txSignerAddress,
+				},
+				Amount: &types.Amount{
+					Value:    "-" + body.Amount.String(),
+					Currency: OasisCurrency,
+				},
+			},
+			&types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: opIndex + 1,
+				},
+				Type:   OpTransfer,
+				Status: m.status,
+				Account: &types.AccountIdentifier{
+					Address: StringFromAddress(body.Account),
+					SubAccount: &types.SubAccountIdentifier{
+						Address: SubAccountEscrow,
+					},
+				},
+				Amount: &types.Amount{
+					Value:    body.Amount.String(),
+					Currency: OasisCurrency,
+				},
+				RelatedOperations: []*types.OperationIdentifier{
+					&types.OperationIdentifier{
+						Index: opIndex,
+					},
+				},
+			},
+		)
+	case staking.MethodReclaimEscrow:
+		var body staking.ReclaimEscrow
+		if err := cbor.Unmarshal(m.tx.Body, &body); err != nil {
+			return fmt.Errorf("malformed body: %w", err)
+		}
+
+		m.ops = append(m.ops,
+			&types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: opIndex,
+				},
+				Type:   OpReclaimEscrow,
+				Status: m.status,
+				Account: &types.AccountIdentifier{
+					Address: m.txSignerAddress,
+				},
+			},
+			&types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: opIndex + 1,
+				},
+				Type:   OpReclaimEscrow,
+				Status: m.status,
+				Account: &types.AccountIdentifier{
+					Address: StringFromAddress(body.Account),
+					SubAccount: &types.SubAccountIdentifier{
+						Address: SubAccountEscrow,
+					},
+				},
+				Metadata: map[string]interface{}{
+					ReclaimEscrowSharesKey: body.Shares.String(),
+				},
+				RelatedOperations: []*types.OperationIdentifier{
+					&types.OperationIdentifier{
+						Index: opIndex,
+					},
+				},
+			},
+		)
+	default:
+		return fmt.Errorf("method not supported")
```

Comment:
(following from other comment) here, if we would instead say, "other transaction types don't result in any operations. do not append anything"